RichString.c

```diff
@@ -21,7 +22,11 @@ in the source distribution for its full text.
 
 #define charBytes(n) (sizeof(CharType) * (n))
 
-static void RichString_extendLen(RichString* this, int len) {
+static void RichString_extendLen(RichString* this, size_t len) {
+   if (len > (SIZE_MAX - 1) / sizeof(CharType)) {
```

Comment:
```suggestion
   if (len > SIZE_MAX / sizeof(CharType) - 1) {
```

Also, why the special handling here? Other places check only for max number of characters, not max number of bytes for storing these characters.

Reply:
> Also, why the special handling here? Other places check only for max number of characters, not max number of bytes for storing these characters.

You don't want the total size in bytes overflow a `size_t` type, do you?
Because of the `+1` on allocated length, I cannot use `xAllocArray` to simplify things.

Reply:
The expression I gave forces `len` to be limited to one byte less as its maximum. With your check, the `-1` is more or less truncated by the `/ sizeof(CharType)` …

Reply:
Acknowledged. My logic error.